### PR TITLE
GODRIVER-1695 always use heartbeatTimeout for heartbeat connection

### PR DESCRIPTION
**Pull request by iwysiu**

### CODE REVIEW

File: x/mongo/driver/topology/server.go [hunk 1]
```diff
@@ -546,7 +546,7 @@ func (s *Server) createConnection() (*connection, error) {
 		WithReadTimeout(func(time.Duration) time.Duration { return s.cfg.heartbeatTimeout }),
 		WithWriteTimeout(func(time.Duration) time.Duration { return s.cfg.heartbeatTimeout }),
 	}
-	opts = append(opts, s.cfg.connectionOpts...)
+	opts = append(s.cfg.connectionOpts, opts...)
```
Comment (divjotarora) [inline x/mongo/driver/topology/server.go:549]: I know the code was already this way, but can we move `WithHandshaker` and `WithMonitor` into the `opts` slice declaration above and have this `append` line be directly after?

Comment (iwysiu) [reply x/mongo/driver/topology/server.go:556]: done

File: x/mongo/driver/topology/server_test.go [hunk 2]
```diff
@@ -312,6 +312,27 @@ func TestServer(t *testing.T) {
...
+			address.Address("localhost"),
+			WithConnectionOptions(func(connOpts ...ConnectionOption) []ConnectionOption {
+				return append(
+					connOpts,
+					WithReadTimeout(func(time.Duration) time.Duration { return socketTimeout }),
+					WithWriteTimeout(func(time.Duration) time.Duration { return socketTimeout }),
+				)
+			}),
+		)
+		require.NoError(t, err)
```
Comment (divjotarora) [inline x/mongo/driver/topology/server_test.go:328]: nit: We should be using the `assert` lib, not `testify/require`.

File: x/mongo/driver/topology/server_test.go [hunk 3]
```diff
@@ -312,6 +312,27 @@ func TestServer(t *testing.T) {
...
+
+		conn, err := s.createConnection()
+		require.NoError(t, err)
+
+		require.Equal(t, s.cfg.heartbeatTimeout, conn.readTimeout, "expected readTimeout to be: %v, got: %v", s.cfg.heartbeatTimeout, conn.readTimeout)
```
Comment (divjotarora) [inline x/mongo/driver/topology/server_test.go:333]: This test would still pass if we incorrectly used a connection's socketTimeout as a server's heartbeatTimeout. Can you add an assertion that `s.cfg.heartbeatTimeout` is 10 seconds?

Comment (iwysiu) [reply x/mongo/driver/topology/server_test.go]: done

File: x/mongo/driver/topology/server_test.go (hunk 2 above)
Comment (iwysiu) [reply x/mongo/driver/topology/server_test.go]: done

**Pull request merged**